cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -305,21 +342,104 @@ const char* trim_utf8(gdv_int64 context, const char* data, gdv_int32 data_len,
     --end;
   }
 
-  // string with no leading/trailing spaces, return original string
-  if (start == 0 && end == data_len - 1) {
-    *out_len = data_len;
-    return data;
+  // string has some leading/trailing spaces and some non-space characters
+  *out_len = end - start + 1;
+  return data + start;
+}
+
+// Trims characters present in the trim text from the left end of the base text
+FORCE_INLINE
+const char* ltrim_utf8_utf8(gdv_int64 context, const char* basetext,
```

Comment:
You need to handle the cases when the input is a malformed utf8 string. Currently the code might give buffer overflow if last character is incomplete multibyte character or the ltrim() might go into an infinite loop if there is an invalid byte in the middle.

Reply:
Added logic which will sniff out an invalid byte or an incomplete multibyte character and will throw an error, as was discussed offline. Also, added unit tests for the same.